timed/reports/views.py

```diff
@@ -80,12 +95,17 @@ class CustomerStatisticViewSet(AggregateQuerysetMixin, ReadOnlyModelViewSet):
     ]
 
     def get_queryset(self):
-        queryset = Report.objects.all()
+        """Don't show statistics to customers."""
+        user = self.request.user
 
+        queryset = Report.objects.all()
         queryset = queryset.values("task__project__customer")
         queryset = queryset.annotate(duration=Sum("duration"))
         queryset = queryset.annotate(pk=F("task__project__customer"))
 
+        if CustomerAssignee.objects.filter(user=user, is_customer=True).exists():
```

Comment:
Deny access by default, only allow if explicitly needed. This code only forbids to customers, but everyone else sees all the data